qtranslate_core.php

```diff
@@ -1138,6 +1124,9 @@ function qtranxf_convertURL( $url = '', $lang = '', $forceadmin = false, $showDe
         $showDefaultLanguage = ! $q_config['hide_default_language'];
     }
     $showLanguage = $showDefaultLanguage || $lang != $q_config['default_language'];
+
+    $url = apply_filters( 'qtranslate_convert_url_before', $url, $lang );
```

Comment:
Can we take this part separately in a different PR? The rest looks fine.

Reply:
ok, I have removed the commit from the PR.
I will submit again after the merge, as it depends on this PR